recipes-devtools/upm/upm_0.7.1.bb

```diff
@@ -16,16 +16,14 @@ inherit distutils-base pkgconfig python-dir cmake
 CFLAGS_append_edison = " -msse3 -mfpmath=sse"
 
 FILES_${PN}-doc += " ${datadir}/upm/examples/"
-RDEPENDS_${PN} += " mraa"
+RDEPENDS_${PN} += " mraa mraa-java"
```

Comment:
@pylbert We need to check if BINDINGS_${PN}-mraa contains "java" before making it a runtime dependency.  If java-mraa is disabled by user, upm will fail also.  So it needs to be something like this:

RDEPENDS_${PN} += "${@bb.utils.contains('BINDINGS_pn-mraa', 'java', 'java-mraa', '', d)}"

@bavery22 is this correct ?